platform/bukkit/plugin/src/main/java/me/lokka30/treasury/plugin/bukkit/TreasuryBukkit.java

```diff
@@ -50,6 +51,16 @@
      */
     @Override
     public void onEnable() {
+        // get rid of this warning: https://img.mrivanplays.com/jaqzdjmvpz.png .We register a
+        // listener for our events for backwards compatibility. Couldn't bukkit be smarter and not
+        // print it for events which are created by the plugin which is listening??
```

Comment:
should I submit an issue to Paper about this?

Reply:
I'm a fan of adding a class annotation for that.

From that example a missing annotation should not throw a CNFE.
https://www.baeldung.com/classnotfoundexception-missing-annotation

So something like that could be possible in Paper-API:

```java
@Retention(RetentionPolicy.RUNTIME)
    @Target(ElementType.TYPE)
    public @interface DoNotWarnFor {
        Class<?>[] plugins();
    }
    
    @Deprecated
    @DoNotWarnFor(plugins = TreasuryBukkit.class)
    public class SomeEvent extends Event {
        
        @Override
        public @NotNull HandlerList getHandlers() {
            return null;
        }
    }
```

That could be filtered.

Reply:
But that is the lazy variant, better would be to compare the class loader of the event class with the class loader of there registering class.

Reply:
Thats just a comment explaining why is that code. What the fuck have you actually did here?